v3-sdk/routing/src/libs/utils.ts

```diff
@@ -0,0 +1,27 @@
+import JSBI from 'jsbi'
+
+export function fromReadableAmount(amount: number, decimals: number): JSBI {
+  const extraDigits = Math.pow(10, countDecimals(amount))
+  const adjustedAmount = amount * extraDigits
+  return JSBI.divide(
+    JSBI.multiply(
+      JSBI.BigInt(adjustedAmount),
+      JSBI.exponentiate(JSBI.BigInt(10), JSBI.BigInt(decimals))
+    ),
+    JSBI.BigInt(extraDigits)
```

Comment:
Could this be 0? I guess so if there are not extra digits 🤔

Reply:
This would always be one, as 10^0 power = 1, and divide by 1 will do nothing 🙂